packages/data-explorer-ui/src/hooks/useAuthentication/useAuthenticationComplete.ts

```diff
@@ -34,7 +35,19 @@ export const useAuthenticationComplete = (
  * @returns path to be used as the initial route history.
  */
 function initRouteHistory(path: string): string {
-  return path === ROUTE_LOGIN ? "/" : path;
+  return path === ROUTE_LOGIN ? "/" : removeInactivityTimeoutQueryParam(path);
+}
+
+/**
+ * Removes the inactivity timeout query parameter from the path.
+ * the inactivity timeout parameter is used to indicate that the session has timed out; remove the parameter to
+ * clear the session timeout banner after the user logs in again.
+ * @param path - Path.
+ * @returns path without the inactivity timeout query parameter.
+ */
+function removeInactivityTimeoutQueryParam(path: string): string {
+  const regex = new RegExp(`\\?${INACTIVITY_PARAM}.*`);
```

Comment:
I wonder if it would work to use a `URLSearchParams` object (e.g. via a `URL` object) to modify the query params. The main benefit of this would be that if there were other query params for some reason, they would be left alone. (There might be other ways to accomplish this, but this is the only one that I know would be readily available.)

If we really expect the inactivity param to only appear alone, though, the only changes I would make would be to really make sure that only the param name we want gets matched, by using our `escapeRegExp` function and making sure there's something marking the end of the name:

```js
const regex = new RegExp(`\\?${escapeRegExp(INACTIVITY_PARAM)}(?:$|[=&].*)`);
```

(This will still match the entire query string when the inactivity param is present, it's just more precise about how it determines that the param is there)

Reply:
I think I went with the "clear anything after the inactivity param" to cover this weird use case:

- user is logged out and the url becomes `/path?inactivityTimeout=true`
- user updates the path with `/path?inactivityTimeout=true&catalog=my-catalog`
- user logs back in

I want to clear everything after `?inactivityTimeout` because the catalog will be in state and after logging back in the catalog will be added to the URL.

Thoughts?

Reply:
Ah, okay. I don't know the details of how query parameters might be used in the app, so I was inclined to be cautious and just remove the one parameter; if we know it's okay to remove all of them, though, the alternative regex I suggested should work for that -- the only change I made was to make it match the parameter name more reliably, so that e.g. it will remove the entirety of `?inactivityTimeout=true&catalog=my-catalog` but will not affect `?inactivityTimeoutExample=true`

Reply:
@hunterckx brilliant thank you! Done!